mediator/esmFldsExchange_coastal_mod.F90

```diff
@@ -236,6 +258,83 @@ subroutine esmFldsExchange_coastal_advt(gcomp, phase, rc)
       deallocate(S_flds)
     end if
 
+    ! ---------------------------------------------------------------------
+    ! to ice: atm fields
+    ! ---------------------------------------------------------------------
+    if (coastal_attr%atm_present .and. coastal_attr%ice_present) then
+      allocate(S_flds(14))
```

Comment:
@SmithJos13 You have 13 field but the size is 14. Please check also other sections. They must be consistent.